Automattic/simplenote-macos
# New Auth Flow: Mk2

### Fix
In this PR we're:

- Implementing **SignupRemote** (borrowing @eshurakov 's code!!)
- Implementing **MagicLinkAuthenticator** (same, Evgeny's code)
- Implementing **Signup Requested UI**
- Wiring all the things together!
- Updating the CI secrets

cc @eshurakov 

---

### Test: Preparation!
1. Update your `mobile-secrets` checkout
2. Delete all of the **Simplenote Mac** apps (Electron / AppStore): they'll interfere with the Magic Link handling
3. Make sure you're logged off
4. Install this branch!

### Test: Running / Background
1. Click over the **Sign Up** button
2. Enter your email
3. Click over the **Sign Up** Button

- [x] Verify that shortly after, the SignUp Requested UI shows up
- [x] Verify that clicking **Back** presents the Authentication UI again
- [x] Verify that if you click on the **SignUp Email Link** (**and complete Signup**) the Simplenote Native App ends up logged in

### Test: Running / Background
1. Click over the **Sign Up** button
2. Enter your email
3. Click over the **Sign Up** Button
4. **Close Simplenote**

- [ ] Verify that if you click on the **SignUp Email Link** (**and complete the Signup flow**) the Simplenote Native App ends up logged in

### Test: Regressions
1. Click over the **Log In with WordPress.com** button

- [x] Verify the WordPress SSO flow still works!

### Release
These changes do not require release notes.
